src/adaptors/mod.rs

```diff
@@ -803,6 +803,88 @@ impl<I, Pred> Iterator for DedupBy<I, Pred>
     }
 }
 
+/// An iterator adaptor that removes repeated duplicates, while keeping a count of how many
+/// repeated elements were present. This will determine equality using a comparison function.
+///
+/// See [`.dedup_by_with_count()`](../trait.Itertools.html#method.dedup_by_with_count) or
+/// [`.dedup_with_count()`](../trait.Itertools.html#method.dedup_with_count) for more information.
+#[must_use = "iterator adaptors are lazy and do nothing unless consumed"]
+pub struct DedupByWithCount<I, Pred>
```

Comment:
Should this be on the `adaptors` module?

Reply:
This location of `DedupByWithCount` is fine. The `adaptors/mod.rs` file is absolutely getting over-sized, though. A restructuring is overdue, but this PR doesn't need to do it.